Libplanet.Net/Transports/NetMQTransport.cs

```diff
@@ -439,9 +446,26 @@ CancellationToken cancellationToken
                 }
                 else
                 {
-                    return new Message[0];
+                    return Array.Empty<Message>();
                 }
             }
+            catch (TaskCanceledException toe) when (timerCts.IsCancellationRequested)
+            {
+                if (returnWhenTimeout)
+                {
+                    return Array.Empty<Message>();
```

Comment:
`returnWhenTimeout` was intended to return received messages by the timeout. For example, if 10 replies were expected and only 5 messages were received when the timeout was fired, 5 messages will be returned when `returnWhenTimeout` is set to true. If `returnWhenTimeout` is false, then throw a timeout exception.

Reply:
Thanks for pointing. then as my understanding, this behavior will only be needed when `returnWhenTimeout` is true. is it right?

https://github.com/planetarium/libplanet/pull/2522/files#diff-87d2edd35fa01273fa9229c93431561886cf57fae9e11593c90c4165a89c2ce7R878-R881

Reply:
Yes. it can look little weird but required for the situation when broadcasting big amount of staged transactions and the timeout occurs while receiving some of them.

Reply:
🆗 I'm going to try to unify those considerations... 🤔

Reply:
@limebell Added https://github.com/planetarium/libplanet/pull/2522/commits/a7efbe1d0fead17b8d8e7e2c2d1f7efdb52e9dd8 to address. PTAL